.github/workflows/ci-post-land.yml

```diff
@@ -0,0 +1,351 @@
+        run: |
+          jq -n \
+            --arg msg "*${{ github.job }}* job in ${{ github.workflow }} workflow failed with $IMAGE_TAG." \
+            --arg url "https://github.com/${{ github.repository }}/actions/runs/${{github.run_id}}" \
+            '{
+              "attachments": [
+                {
+                  "text": $msg,
+                  "actions": [
+                    {
+                      "type": "button",
+                      "text": "Visit Job",
+                      "url": $url
+                    }
+                  ]
+                }
+              ]
+            }' > /tmp/payload
+          curl -X POST -H 'Content-type: application/json' -d @/tmp/payload ${{ secrets.WEBHOOK_PUSH }}
+
+  build-release-images:
+    runs-on: ubuntu-latest-xl
+    continue-on-error: false
+    env:
+      TAG: github-1
+      CHANGES_TARGET_BRANCH: ${{ needs.prepare.outputs.changes-target-branch }}
+    environment:
+      name: Docker
+      url: https://hub.docker.com/u/diem
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0 #get all the history!!!
+      - name: setup_aws_ecr_login
+        run: |
+          echo 'AWS_ECR_ACCOUNT_URL=${{ secrets.ENV_NOVI_ECR_AWS_ACCOUNT_NUM }}.dkr.ecr.${{ secrets.ENV_NOVI_ECR_AWS_REGION }}.amazonaws.com' >> $GITHUB_ENV
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.ENV_NOVI_ECR_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ENV_NOVI_ECR_AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.ENV_NOVI_ECR_AWS_REGION }}
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1.3.3
+      - name: Sign in to dockerhub, install image signing cert.
+        uses: ./.github/actions/dockerhub_login
+        with:
+          username: ${{ secrets.ENV_DOCKERHUB_USERNAME }}
+          password: ${{ secrets.ENV_DOCKERHUB_PASSWORD }}
+          key_material: ${{ secrets.ENV_DOCKERHUB_KEY_MATERIAL }}
+          key_name: ${{ secrets.ENV_DOCKERHUB_KEY_NAME }}
+          key_password: ${{ secrets.ENV_DOCKERHUB_KEY_PASSWORD }}
+      - name: should pre build docker images (targeting a release branch)?
+        if: ${{ github.ref == 'refs/heads/auto' }}
+        run: |
+          if  [[ ! "$CHANGES_TARGET_BRANCH" =~ "^release-[0-9|.]+$" ]] && [[ ! "$CHANGES_TARGET_BRANCH" =~ "^gha-test-[0-9|.]+$" ]] ; then
+            echo Targeting branch $CHANGES_TARGET_BRANCH will not pre-publish docker images.
+          fi
+      - name: pre-release docker images
+        if: ${{ github.ref == 'refs/heads/auto' }}
+        run: |
+          BRANCH="$CHANGES_TARGET_BRANCH"
+          success=0
+          tmpfile=$(mktemp)
+          echo "Failed to push:" > "${tmpfile}"
+          docker/build_push.sh -u -p -b ${BRANCH} -n client || success=$(echo "client" >> "${tmpfile}"; echo 1)
```

Comment:
nit: would be cleaner to do as a `foreach item in "client init faucet tools validator validator-tcb cluster-test" ` loop.

Reply:
I did write this bit, and part of the reason for not running a loop was to let people copy paste the commands to perform a local release if they had the secrets and needed to release for some reason (ci down, etc).   Otherwise agree.